modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxManager.java

```diff
@@ -149,4 +149,19 @@ CompletableFuture<Void> cleanup(
      * @return Future of all read-only transactions with read timestamp less or equals the given new low watermark.
      */
     CompletableFuture<Void> updateLowWatermark(HybridTimestamp newLowWatermark);
+
+    /**
+     * Registers the infligh update for a transaction.
+     *
+     * @param tx The transaction id.
+     * @return {@code True} if the inflight was registered. The update must be failed on false.
+     */
+    boolean addInflight(UUID txId);
```

Comment:
I'd say that add/removeInflight should be a part of InternalTransaction and not TxManager.
Besides that javadoc requires clarification. And what's more important sql currently totally miss the thing.

Reply:
I disagree. It makes sense only for RW transactions, so it conflicts with RO implementation.